addons/spreadsheet/static/tests/currency/currency_test.js

```diff
@@ -72,3 +75,26 @@ QUnit.test("Currency rates are loaded once by clock", async (assert) => {
     await waitForDataSourcesLoaded(model);
     assert.verifySteps(["FETCH:2"]);
 });
+
+QUnit.test("Currency rates per date", async (assert) => {
+    const model = await createModelWithDataSource({
+        mockRPC: async function (route, args) {
+            if (args.method === "get_rates_for_spreadsheet") {
+                assert.step("FETCH");
```

Comment:
Maybe also check that we send the correct dates as argument while we're at it ? PR seems good otherwie